include/EclipseMonitor/Eth/TransactionMgr.hpp

```diff
@@ -19,57 +18,120 @@ namespace Eth
 {
 
 
-enum class TnxVersion : uint8_t
+enum class TxnVersion : uint8_t
 {
 	Legacy  = 0,
-	Ver0x01 = 1,
-	Ver0x02 = 2,
-} // enum class TnxVersion
+	AccessList = 1,
+	DynamicFee = 2,
+}; // enum class TnxVersion
 
 
 class TransactionMgr
 {
 public: // static members:
 
-	static std::tuple<TnxVersion, Internal::Rlp::List> ParseTnx(
-		const Internal::Rlp::Bytes& rlpBytes
-	); // TODO[Tuan]
+	static std::tuple<TxnVersion, Internal::Obj::Object> ParseTnx(
+		const Internal::Obj::Bytes& rlpBytes
+	)
+	{
+		TxnVersion version;
+		uint8_t firstByte = rlpBytes[0];
+		Internal::Obj::Bytes innerRlp;
+		if (firstByte == 0x01)
+		{
+			version = TxnVersion::AccessList;
+			innerRlp = Internal::Obj::Bytes(rlpBytes.data() + 1, rlpBytes.data() + rlpBytes.size());
+		}
+		else if (firstByte == 0x02)
+		{
+			version = TxnVersion::DynamicFee;
+			innerRlp = Internal::Obj::Bytes(rlpBytes.data() + 1, rlpBytes.data() + rlpBytes.size());
+		}
+		else
+		{
+			version = TxnVersion::Legacy;
+			innerRlp = rlpBytes;
+		}
+
+		Internal::Obj::Object txObj
+			= Internal::Rlp::ParseRlp(innerRlp.GetVal());
+		return std::make_tuple(version, txObj);
+	}
 
 
 	static TransactionMgr FromBytes(
-		const Internal::Rlp::Bytes& rlpBytes
+		const Internal::Obj::Bytes& rlpBytes
 	)
 	{
-		TnxVersion ver;
-		Internal::Rlp::List body;
-		std::tie(ver, body) = ParseTnx(rlpBytes);
+		TxnVersion ver;
+		Internal::Obj::Object txnObj;
+		std::tie(ver, txnObj) = ParseTnx(rlpBytes);
+
+		Internal::Obj::ListBaseObj &txnListObj = txnObj.AsList();
 
-		return TransactionMgr(ver, body);
+		return TransactionMgr(ver, txnObj, txnListObj);
 	}
 
+
 public:
 
 	TransactionMgr(
-		TnxVersion version,
-		const Internal::Rlp::List& body
+		TxnVersion version,
+		const Internal::Obj::Object& txnObj,
+		Internal::Obj::ListBaseObj& txnListObj
 	) :
 		m_version(version),
-		m_tnxBody(body)
-	{};
+		m_txnObj(txnObj),
+		m_txnBody(txnListObj)
```

Comment:
for var members that require more work in the constructor's body, initialize them first in the initialization list using their default constructors (e.g., `m_data()`). (btw, when adding them to the initialization list, ensure they're in the same order as they declared in line 128)